### DISCUSSION TITLE
pull_request workflow not triggering on feature branch

### DISCUSSION THREAD

**Discussion by rahulrane50** [community/community#31198] [Actions]
I have created CI yaml file in default master branch which sets to trigger bunch of tests if any pull_request is created against master branch or 'foo-dev/**'. Now I'm expecting that if I create a pull request against 'foo-dev/bar-feature-branch.0' then CI workflow should get triggered but that's not happening. 

My yaml file :

`on:
  push:
    branches:
      - master
      - 'foo-dev/**'
  pull_request:
    branches:
      - master
      - 'foo-dev/**'
`
It worked only when I copy pasted this ci.yaml file from master branch to 'foo-dev/bar-feature-branch.0' branch. I even tried having branches in array style and adding escape character before '/' but still didn't work when kept it only in master branch. Can someone please help me understand what did I miss?

**Comment by Simran-B** [comment]
So you open a PR with `foo-dev/bar-feature-branch.0` as the base branch, but this branch does not contain the workflow, at least not the version in the master branch? That's likely the problem because:

> Each workflow run will use the version of the workflow that is present in the associated commit SHA or Git ref of the event.

https://docs.github.com/en/actions/using-workflows/triggering-a-workflow
